.github/workflows/automation-trigger-test.yml

```diff
@@ -1,18 +1,117 @@
-#on: [push]
+name: SaaS SDK Automation Tests
 on:
  workflow_dispatch:
+ pull_request:
  
 jobs:
-  build:
-    runs-on: ubuntu-latest
+  # This job creates a static lib which will be used by Qentelli Automation Testbed App. Its uploaded as an artifact.
+  build-Branch-Static-framework:
+    runs-on: macos-latest
     steps:
-    - uses: actions/github-script@v6
-      with:
-        github-token: ${{ secrets.BRANCHLET_ACCESS_TOKEN_PUBLIC  }}
-        script: |
-          await github.rest.actions.createWorkflowDispatch({
-            owner: 'BranchMetrics',
-            repo: 'qentelli-saas-sdk-testing-automation',
-            workflow_id: 'ios-maven.yml',
-            ref: 'master'
-          })
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Build xcframework
+        run: |
+            ./scripts/prep_static_xcframework.sh
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v3
```

Comment:
For the purpose of sharing data between jobs - unless there's a need to save/persist data after a workflow run, we can use cache via `actions/cache@v3` in place of upload and downloading artifacts.Using cache is in general faster than uploading and downloading artifacts.

Reply:
@JagadeeshKaricherla-branch As per my understanding, Cache will retrieve same files. Here files are changing with every run. Cache is used for re-using same files between different runs, so they are faster.  For example,  we are using cache here - https://github.com/BranchMetrics/ios-branch-deep-linking-attribution/blob/master/.github/workflows/release.yml for faster building of pods. Here it will not have any impact because they are re-generated with every run. Correct me if I am wrong.